tests/foreman/ui/test_contentview.py

```diff
@@ -1830,115 +1834,406 @@ def test_positive_restart_publish_via_dynflow(self):
 
+        @assert: User with read-only role for content view can view the
+        repository in the content view
 
         @CaseLevel: Integration
         """
+        repo_name = gen_string('alpha')
+        cv_name = gen_string('alpha')
+        user_login = gen_string('alpha')
+        user_password = gen_string('alphanumeric')
+        role_name = gen_string('alpha')
+        # create a role with content views read only permissions
+        role = entities.Role(name=role_name).create()
+        entities.Filter(
+            organization=[self.organization],
+            permission=entities.Permission(
+                resource_type='Katello::ContentView').search(
+                filters={'name': 'view_content_views'}),
+            role=role,
+            search=None
+        ).create()
+        # create read only products permissions and assign it to our role
+        entities.Filter(
+            organization=[self.organization],
+            permission=entities.Permission(
+                resource_type='Katello::Product').search(
+                filters={'name': 'view_products'}),
+            role=role,
+            search=None
+        ).create()
+        # create a user and assign the above created role
+        entities.User(
+            default_organization=self.organization,
+            organization=[self.organization],
+            role=[role],
+            login=user_login,
+            password=user_password
+        ).create()
+        # create a content view with the main admin account
+        with Session(self.browser) as session:
+            # create a repository
+            # note: the user created above has not access to repositories
+            self.setup_to_create_cv(repo_name=repo_name)
+            # create the first content view
+            make_contentview(
+                session, org=self.organization.name, name=cv_name)
+            self.assertIsNotNone(self.content_views.search(cv_name))
+            # add repository to the created content view
+            self.content_views.add_remove_repos(cv_name, [repo_name])
+        # login as the user created above
+        with Session(self.browser, user_login, user_password) as session:
+            session.nav.go_to_content_views()
+            # assert the user can view the content view
+            cv_element = self.content_views.search(cv_name)
+            self.assertIsNotNone(cv_element)
+            # assert that the user can view the repo in content view
+            self.content_views.click(cv_element)
+            # cv_element.click()
+            self.content_views.wait_for_ajax()
+            self.content_views.click(tab_locators.contentviews.tab_content)
+            self.content_views.click(locators.contentviews.content_repo)
+            self.content_views.text_field_update(
+                locators.contentviews.repo_search, repo_name)
+            self.assertIsNotNone(self.content_views.wait_until_element(
```

Comment:
as I suggested for previous test case, you can check that current user cannot create new CV, that necessary button is absent on a page to prove that you can only view CVs